### PR TITLE
Set integration with SPI in preview mode

### PR DESCRIPTION
**Pull request by Michkov**
- Allow to configure SPI integration using preview mode
- Add script hack/spi/oauth-flow.sh to proceed oauth workflow

### CODE REVIEW

**Comment by skabashnyuk** [comment]
@Michkov Can you explain the purpose of this pr. For what and how it should be used?

**Comment by Michkov** [comment]
Preview mode is for local testing

**Comment by skabashnyuk** [comment]
@Michkov I'm sorry I don't understand the relation between preview mode and `hack/spi/oauth-flow.sh` script.

CC @sbose78

**Comment by Michkov** [comment]
Let's split it into two parts:

1. I want to provide developers easy way to setup SPI. That's done via variables, they will set them once and it will work for all deployments.
2. In #65 there was mention how to test, so I automated it so developers can verify that it's set properly.

Taking point 2) as nice to have, but if you have different opinion, then I can remove it from the PR.

**Comment by skabashnyuk** [comment]
`hack/spi/oauth-flow.sh` It is a real hack that is not suppose to be widely used. I prefer to keep it only in spi repository. Can we remove it from this pr?

**Comment by Michkov** [comment]
@skabashnyuk 'a real hack' was removed from hack folder :-)

File: hack/preview-template.env [hunk 1]
```diff
@@ -11,3 +11,10 @@ export MY_GIT_FORK_REMOTE=
 export MY_GITHUB_ORG=
 ### Personal API token with repo and delete_repo permission
 export MY_GITHUB_TOKEN=
+
+## SPI integration
+### Based on https://github.com/redhat-appstudio/service-provider-integration-operator#configuration
+export SHARED_SECRET= # Random string
+export SPI_TYPE=
```
Comment (skabashnyuk) [inline hack/preview-template.env:18]: # Github

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**